### ISSUE TITLE
addcounts! using radix sort doesn't add counts

### ISSUE THREAD

**Issue by awllee** [JuliaStats/StatsBase.jl#376]
Instead it gives the counts of any elements in the new array as well the previous counts of any elements that do not appear in the new array.

I believe this is because of the initialization`tmpcount = 1` in `_addcounts_radix_sort_loop!` rather than using the previous value + 1 [if it exists].

Example:

```
using StatsBase
dict = Dict{Int64, Int64}()
vals1 = [1 ; 2 ; 3]
vals2 = [1 ; 2 ; 4]
StatsBase.addcounts!(dict, vals1)
println(dict)
StatsBase.addcounts!(dict, vals2)
println(dict)
```

gives as output:

```
Dict(2=>1,3=>1,1=>1)
Dict(4=>1,2=>1,3=>1,1=>1)
```

**Comment by nalimilan** [comment]
Good catch! If you think you have found the cause of the bug, a pull request (with a test) would be welcome.

Cc: @xiaodaigh

**Comment by xiaodaigh** [comment]
ok looking into it
